lib/actions/local/configure.coffee

```diff
@@ -42,15 +45,133 @@ CONFIGURATION_SCHEMA =
 			fileset: true
 			type: 'ini'
 			location:
-				path: 'system-connections'
-				partition:
-					primary: 1
+				path: CONNECTIONS_FOLDER.slice(1)
+				partition: BOOT_PARTITION
 		config_json:
 			type: 'json'
 			location:
 				path: 'config.json'
-				partition:
-					primary: 1
+				partition: BOOT_PARTITION
+
+inquirerOptions = (data) -> [
+	{
+		message: 'Network SSID'
+		type: 'input'
+		name: 'networkSsid'
+		default: data.networkSsid
+	}
+	{
+		message: 'Network Key'
+		type: 'input'
+		name: 'networkKey'
+		default: data.networkKey
+	}
+	{
+		message: 'Do you want to set advanced settings?'
+		type: 'confirm'
+		name: 'advancedSettings'
+		default: false
+	}
+	{
+		message: 'Device Hostname'
+		type: 'input'
+		name: 'hostname'
+		default: data.hostname,
+		when: (answers) ->
+			answers.advancedSettings
+	}
+	{
+		message: 'Do you want to enable persistent logging?'
+		type: 'confirm'
+		name: 'persistentLogging'
+		default: data.persistentLogging
+		when: (answers) ->
+			answers.advancedSettings
+	}
+]
+
+getConfiguration = (data) ->
+	_ = require('lodash')
+	inquirer = require('inquirer')
+
+	# `persistentLogging` can be `undefined`, so we want
+	# to make sure that case defaults to `false`
+	data = _.assign data,
+		persistentLogging: data.persistentLogging or false
+
+	inquirer.prompt(inquirerOptions(data))
+	.then (answers) ->
+		return _.merge(data, answers)
+
+# Taken from https://goo.gl/kr1kCt
+CONNECTION_FILE = '''
+	[connection]
+	id=resin-wifi
+	type=wifi
+
+	[wifi]
+	hidden=true
+	mode=infrastructure
+	ssid=My_Wifi_Ssid
+
+	[wifi-security]
+	auth-alg=open
+	key-mgmt=wpa-psk
+	psk=super_secret_wifi_password
+
+	[ipv4]
+	method=auto
+
+	[ipv6]
+	addr-gen-mode=stable-privacy
+	method=auto
+'''
+
+prepareConnectionFile = (target) ->
+	_ = require('lodash')
+	imagefs = require('resin-image-fs')
+
+	imagefs.listDirectory
+		image: target
+		partition: BOOT_PARTITION
+		path: CONNECTIONS_FOLDER
+	.then (files) ->
+		# The required file already exists
+		if _.includes(files, 'resin-wifi')
+			return null
+
+		# Fresh image, new mode, accoding to https://github.com/resin-os/meta-resin/pull/770/files
+		if _.includes(files, 'resin-sample.ignore')
+			return imagefs.copy
+				image: target
+				partition: BOOT_PARTITION
+				path: "#{CONNECTIONS_FOLDER}/resin-sample.ignore"
+			,
+				image: target
+				partition: BOOT_PARTITION
+				path: "#{CONNECTIONS_FOLDER}/resin-wifi"
+			.thenReturn(null)
+
+		# Legacy mode, to be removed later
```

Comment:
Well, I think it's OK for CLI v4 to stop support resin OS < 2.1 or something like that. But indeed it won't happen too soon.